modules/3d/include/opencv2/3d/ptcloud.hpp

```diff
@@ -11,6 +11,334 @@ namespace cv {
+    {
+        return confidence;
+    }
+
+    //! Set the number of models expected.
+    inline void setNumberOfModelsExpected(int number_of_models_expected_)
+    {
+        number_of_models_expected = number_of_models_expected_;
+    }
+
+    //! Get the expected number of models.
+    inline int getNumberOfModelsExpected() const
+    {
+        return number_of_models_expected;
+    }
+
+    /**
+     * @brief Set the number of threads to be used.
+     *
+     * @param number_of_threads_ The number of threads to be used.
+     * (0 sets the value automatically, a negative number turns parallelization off)
+     *
+     * @note Not all SAC methods have a parallel implementation. Some will ignore this setting.
+     */
+    inline void setNumberOfThreads(int number_of_threads_)
+    {
+        number_of_threads = number_of_threads_;
+    }
+
+    // Get the number of threads to be used.
+    inline int getNumberOfThreads() const
+    {
+        return number_of_threads;
+    }
+
+    //! Set state used to initialize the RNG(Random Number Generator).
+    inline void setRandomGeneratorState(uint64 rng_state_)
+    {
+        rng_state = rng_state_;
+    }
+
+    //! Get state used to initialize the RNG(Random Number Generator).
+    inline uint64 getRandomGeneratorState() const
+    {
+        return rng_state;
+    }
+
+    //! Set custom model coefficient constraint function
+    inline void setCustomModelConstraints(const ModelConstraintFunction &custom_model_constraints_)
+    {
+        custom_model_constraints = custom_model_constraints_;
+    }
+
+    //! Get custom model coefficient constraint function
+    const ModelConstraintFunction &getCustomModelConstraints() const
+    {
+        return custom_model_constraints;
+    }
+
+    /**
+     * @brief Execute segmentation using the sample consensus method.
+     *
+     * @param input_pts Original point cloud, vector of Point3 or Mat of size Nx3/3xN.
+     * @param[out] labels The label corresponds to the model number, 0 means it
+     * does not belong to any model, range [0, Number of final resultant models obtained].
+     * @param[out] models_coefficients The resultant models coefficients.
+     * Currently supports passing in cv::Mat. Models coefficients are placed in a matrix of NxK,
+     * where N is the number of models and K is the number of coefficients of one model.
+     * The coefficients for each model refer to the comments inside enumeration type SacModelType.
+     * @return Number of final resultant models obtained by segmentation.
+     */
+    int
+    segment(InputArray input_pts, OutputArray labels, OutputArray models_coefficients = noArray());
+
+protected:
+
+    //! The type of sample consensus model used.
+    SacModelType sac_model_type;
+
+    //! The type of sample consensus method used.
+    SacMethod sac_method;
+
+    //! Considered as inlier point if distance to the model less than threshold.
+    double threshold;
```

Comment:
Fields are not allowed in OpenCV public API.

Use `PImpl` pattern + setters/getters

See also: https://github.com/opencv/opencv/wiki/Coding_Style_Guide

Reply:
fixed.